consensus/src/quorum_store/utils.rs

```diff
@@ -250,30 +249,15 @@ impl ProofQueue {
     }
 
     fn remaining_txns_without_duplicates(&self) -> u64 {
-        // All the batch keys for which batch_to_proof is not None. This is the set of unexpired and uncommitted proofs.
-        let unexpired_batch_keys = self
-            .batch_to_proof
-            .iter()
-            .filter(|(_, proof)| proof.is_some())
-            .map(|(batch_key, _)| batch_key)
-            .collect::<HashSet<_>>();
-        let mut remaining_txns = self
-            .txn_summary_to_batches
-            .iter()
-            .filter(|(_, batches)| {
-                batches
-                    .iter()
-                    .any(|batch_key| unexpired_batch_keys.contains(batch_key))
-            })
-            .count() as u64;
+        let mut remaining_txns = self.txn_summary_num_occurrences.len() as u64;
 
         // If a batch_key is not in batches_with_txn_summary, it means we've received the proof but haven't receive the
         // transaction summary of the batch from batch coordinator. Add the number of txns in the batch to remaining_txns.
         remaining_txns += self
```

Comment:
Why do we need to add these? Isn't `self.txn_summary_num_occurrences.len() as u64` enough?

Reply:
We may not have received txn summaries for all batches. It's possible we have a proof but not the corresponding batch. For those proofs, we need to add proof.num_txns to remaining_txns.